pgxn/neon/control_plane_connector.c

```diff
@@ -501,9 +507,16 @@ HandleCreateDb(CreatedbStmt *stmt)
 
 	entry->type = Op_Set;
 	if (downer && downer->arg)
-		entry->owner = get_role_oid(defGetString(downer), false);
+        {
+		const char *owner_name = defGetString(downer);
+		if (RoleIsNeonSuperuser(owner_name))
+			elog(ERROR, "can't create a database with owner neon_superuser");
+		entry->owner = get_role_oid(owner_name, false);
+        }
 	else
+        {
 		entry->owner = GetUserId();
+        }
```

Comment:
Whitespace is pretty off here. spaces/tabs

Reply:
Fixed